app/verification.py

```diff
@@ -32,58 +34,203 @@ def initialize():
         p.execute()
 
 
-def get_verification_methods(appid: str):
-    segments = appid.split(".")
+def _matches_prefixes(appid: str, *prefixes) -> bool:
+    for prefix in prefixes:
+        if appid.startswith(prefix + "."):
+            return True
+    return False
+
+
+def _get_github_username(appid: str) -> str:
+    if _matches_prefixes(appid, "com.github", "io.github"):
+        return appid.split(".")[2]
+    else:
+        return None
+
 
-    if len(segments) < 3:
+def _get_domain_name(appid: str) -> str:
+    if _matches_prefixes(appid, "com.github", "com.gitlab"):
+        # These app IDs are common, and we don't want to confuse people by saying they can put a file on GitHub/GitLab's
+        # main website.
+        return None
+    elif _matches_prefixes(appid, "io.github", "io.gitlab"):
+        # You can, however, verify by putting a file on your *.github.io or *.gitlab.io site
+        return ".".join(reversed(appid.split(".")[0:3]))
+    else:
+        return ".".join(reversed(appid.split(".")[0:2]))
```

Comment:
is `[0:2]` right here, or should it be `[0:-1]` ?

Reply:
Some app IDs are more than 3 segments, and I think usually we want "the first two" rather than "all but the last." For example, org.gnome.Shumate.Demo.

Reply:
I worry about heuristics like this - for now it's probably acceptable, but longer term we're going to need something a little more controllable.  Another heuristic option might be "All segments which don't start with a capital letter" but again, it's just a heuristic, not something necessarily great.

One example of how this will fail with non-fancy app-ids would be a UK developer whose domain is a `.co.uk` - e.g. `uk.co.mangobrain.Infector`